src/Compilers/Core/Portable/VersionHelper.cs

```diff
@@ -74,35 +74,50 @@ private static bool TryParse(string s, bool allowWildcard, ushort maxValue, out
 
             if (hasWildcard)
             {
-                int seconds = ((int)(DateTime.Now.TimeOfDay.TotalSeconds)) / 2;
-                if (seconds > (int)maxValue)
+                for (int i = lastExplicitValue; i < values.Length; i++)
```

Comment:
Let's assert at the beginning of this function that, if allowWildcard is true, then maxValue <  ushort.MaxValue.